affirm/src/main/java/com/affirm/android/AffirmPlugins.java

```diff
@@ -0,0 +1,271 @@
+    private Gson mGson;
+
+    AffirmPlugins(@NonNull Affirm.Configuration configuration) {
+        mConfiguration = configuration;
+    }
+
+    static void initialize(@NonNull Affirm.Configuration configuration) {
+        AffirmPlugins.set(new AffirmPlugins(configuration));
+    }
+
+    private static void set(@NonNull AffirmPlugins plugins) {
+        synchronized (LOCK) {
+            if (mInstance != null) {
+                throw new IllegalStateException("AffirmPlugins is already initialized");
+            }
+            mInstance = plugins;
+        }
+    }
+
+    public static AffirmPlugins get() {
+        synchronized (LOCK) {
+            return mInstance;
+        }
+    }
+
+    static void reset() {
+        synchronized (LOCK) {
+            mInstance = null;
+        }
+    }
+
+    String publicKey() {
+        return mConfiguration.publicKey;
+    }
+
+    String merchantName() {
+        return mConfiguration.merchantName;
+    }
+
+    Affirm.Environment environment() {
+        return mConfiguration.environment;
+    }
+
+    String environmentName() {
+        return mConfiguration.environment.name();
+    }
+
+    String baseUrl() {
+        return mConfiguration.environment.baseUrl;
+    }
+
+    String trackerBaseUrl() {
+        return mConfiguration.environment.trackerBaseUrl;
+    }
+
+    synchronized Gson gson() {
+        if (mGson == null) {
+            mGson =
+                    new GsonBuilder()
+                            .registerTypeAdapterFactory(AffirmAdapterFactory.create()).create();
+        }
+        return mGson;
+    }
+
+    synchronized AffirmHttpClient restClient() {
+        if (mRestClient == null) {
+            OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
+            //add it as the first interceptor
+            clientBuilder.interceptors().add(0, new Interceptor() {
+                @Override
+                @NonNull
+                public Response intercept(@NonNull Chain chain) throws IOException {
+                    final Request request = chain.request()
+                            .newBuilder()
+                            .addHeader("Accept", "application/json")
+                            .addHeader("Content-Type", "application/json")
+                            .addHeader("Affirm-User-Agent", "Affirm-Android-SDK")
+                            .addHeader("Affirm-User-Agent-Version", BuildConfig.VERSION_NAME)
+                            .build();
+
+                    return chain.proceed(request);
+                }
+            });
+            clientBuilder.connectTimeout(5, TimeUnit.SECONDS);
+            clientBuilder.readTimeout(30, TimeUnit.SECONDS);
+            clientBuilder.followRedirects(false);
+
+            mRestClient = AffirmHttpClient.createClient(enableTls12OnPreLollipop(clientBuilder));
+        }
+        return mRestClient;
+    }
+
+    /**
+     * Enables TLS v1.2 when creating SSLSockets.
```

Comment:
As mentioned before, this is not necessary when we bump up the min sdk version to 21